docs/rtk-query/api/created-api/api-slice-utils.mdx

```diff
@@ -105,6 +110,43 @@ dispatch(api.endpoints.getPostById.initiate(1))
 dispatch(api.endpoints.getPostById.initiate(1, { ...options }))
 ```
 
+### `upsertQueryData`
+
+#### Signature
+
+```ts no-transpile
+const upsertQueryData = <T>(
+  endpointName: string,
+  args: any,
+  newEntryData: T
+) => ThunkAction<Promise<CacheEntry<T>>, PartialState, any, AnyAction>;
+```
+
+- **Parameters**
+  - `endpointName`: a string matching an existing endpoint name
+  - `args`: an argument matching that used for a previous query call, used to determine which cached dataset needs to be updated
+  - `newEntryValue`: the value to be written into the corresponding cache entry's `data` field
+
+#### Description
+
+A Redux thunk action creator that, when dispatched, acts as an artificial API request to upsert a value into the cache.
+
+The thunk action creator accepts three arguments: the name of the endpoint we are updating (such as `'getPost'`), the appropriate query arg values to construct the desired cache key, and the data to upsert.
+
+If no cache entry for that cache key exists, a cache entry will be created and the data added. If a cache entry already exists, this will _overwrite_ the existing cache entry data.
```

Comment:
If the value being upserted is `undefined`, will it remove the cache key entirely? If not, is there a way to remove a cache entry entirely? (ie, like `invalidateTags`, but without triggering a refetch)

Reply:
@ggaabe : no, it would replace the `.data` field with `undefined`.   It's the value, not the key.

There isn't currently a way to remove one, but a few folks have asked for something like this.  Could you comment over in #2820 and describe when and why you might want to do that?